vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -1019,18 +1015,21 @@ import { ItemCache } from '@vaadin/grid/src/vaadin-grid-data-provider-mixin.js';
         }));
       }));
 
-      function _fireClickEvent(event, eventName) {
-        if (grid.$connector.activeItem) {
-          event.itemKey = grid.$connector.activeItem.key;
-          const eventContext = grid.getEventContext(event);
-          // if you have a details-renderer, getEventContext().column is undefined
-          if (eventContext.column) {
-            event.internalColumnId = eventContext.column._flowId;
-          }
-          grid.dispatchEvent(new CustomEvent(eventName,
-            { detail: event }));
+    function _fireClickEvent(event, eventName) {
+      const target = event.target;
+      const eventContext = grid.getEventContext(event);
+      const section = eventContext.section;
+
+      if (eventContext.item && !isFocusable(target) && section !== 'details') {
```

Comment:
Wouldn't `section !== details` mean that you'd no longer get an ItemClick event if you click the row details? Sounds like a breaking change if that's the case.

Reply:
The previous workaround relied on the `cell-activate` event to determine which item to send an item click for. The `cell-activate` event it not dispatched when clicking details: https://github.com/vaadin/web-components/blob/master/packages/grid/src/vaadin-grid-active-item-mixin.js#L60-L62

There is also a Flow IT that covers that item-click is not triggered when clicking details.

Reply:
Ah, got it